Add UDF names to the new `used_executable_user_defined_functions` and `used_sql_user_defined_functions` columns in the `system.query_log`

nibblerenush opened the pull request:
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
User Defined Functions (UDF) names are not added to the `system.query_log` table, unlike other functions. This PR implements the addition of the UDF name to one of the two columns `used_executable_user_defined_functions` or `used_sql_user_defined_functions` if the UDF was used in the request.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)
- Motivation: other functions are logged, but the UDF is not.

nibblerenush committed:
Log sql udf names

Add SQL UDF names to the used_functions column in the system.query_log

vitlibar reviewed (changes requested):
Let's check the test.

nibblerenush commented on tests/queries/0_stateless/03525_sql_udf_names_in_used_functions_column.sh:
```diff
@@ -0,0 +1,60 @@
+}
+
+test_sql_udf_single() {
+    check_sql_udf_functions ${FUNCNAME}
```
It is predefined bash variable like RANDOM.
From docs:
"**FUNCNAME**: An array variable containing the names of all shell functions currently in the execution call stack. The element with index 0 is the name of any currently-executing shell function."

nibblerenush committed:
Log UDF names

Two different columns have been created in system.query_log:
- used_executable_user_defined_functions
- used_sql_user_defined_functions
These columns log the executable and sql UDF names respectively

nibblerenush committed:
Fix style check

nibblerenush committed:
Add no-parallel tag

nibblerenush committed:
Generate query_id from generateUUIDv4 function

nibblerenush committed:
Specify query_log for flush

nibblerenush committed:
Add docs

vitlibar commented on tests/queries/0_stateless/03525_sql_udf_names_in_system_query_log.sh:
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# Tags: no-parallel
```
Why is it `no-parallel`? Just use random-generated names for functions and you can run this test in parallel without any troubles.

vitlibar replied:
> But then I won't be able to specify the correct test data in the .reference file, because the function names will be random.

You can replace random names in ClickHouse output with something non-random before checking the output with `.reference` file.

Algunenano replied:
You can use the database name and it will be replaced automatically to `default` for the reference

nibblerenush replied:
Thanks for the help! I created a PR so that the test could run in parallel: https://github.com/ClickHouse/ClickHouse/pull/81284
